### PR TITLE
#365 New submission page

### LINKED ISSUE #365: Add SubmissionNEW page

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #365 

### Changes
- Create new `ApplicationSubmission` page container that receives the structure and only display the page - I wasn't sure if that might be just a component - since there isn't much logic going on here. But I guess we might have later on.
- Change type **FullStructure.info** to also keep the `submissionMessage` used in the Submission page for an application
- Change type **FullStructure.stages** to be replaced with what was agreed on the new [Naming Changes Structure Diagram](https://app.diagrams.net/#G1wEk_eH1uWd8LC0JeKAM58bUkGncxKDgW). 
- Disclaimer: Although changes were done to same structure in use by old Application pages, I didn't have to change anything since the previous `stages` with `submissionMessage` was stored in a different object with type `ApplicationStages` that hasn't changes. And the addition to `submissionMessage` in the `ApplicationDetails` is using the temporary set to be optional field, so not required for previous use.
- Set redirect to the `ApplicationSummary` page when the application latest status isn't SUBMITTED/RE-SUBMITTED
- Links to Application list (based on **application.type**) and home page from the Submission page.

### CODE REVIEW

File: src/containers/Application/ApplicationSubmissionNEW.tsx
Comment (CarlosNZ) [inline src/containers/Application/ApplicationSubmissionNEW.tsx:41]: Do you think we should evaluate this in `useLoadApplication`, to keep more logic out of this display component?

File: src/containers/Application/ApplicationSubmissionNEW.tsx
Comment (CarlosNZ) [inline src/containers/Application/ApplicationSubmissionNEW.tsx:89]: I'm going to assume this UI is basically identical to the old one?

**Review by CarlosNZ — COMMENTED** [review]
Great! One small suggestion about moving the evaluation, see if you think it's a good idea.

File: src/containers/Application/ApplicationSubmissionNEW.tsx
Comment (nmadruga) [reply src/containers/Application/ApplicationSubmissionNEW.tsx:89]: Yep. It is : )

File: src/containers/Application/ApplicationSubmissionNEW.tsx
Comment (nmadruga) [reply src/containers/Application/ApplicationSubmissionNEW.tsx:41]: I'll have a thought tomorrow, I think it's a good idea, but wonder if should do the same for `startMessage` in `useLoadTemplate` => In PR #404 - XD

Comment (CarlosNZ) [reply src/containers/Application/ApplicationSubmissionNEW.tsx:41]: Yes, I actually just made the same comment on that one too -- haven't submitted review yet though.

**Review by CarlosNZ — APPROVED** [review]
LGTM -- once you merge this I'll pull it into my #409 PR and make sure that's working properly with it.

You want to do the change to move the Start/Submit message evaluation to hooks?

**Comment by nmadruga** [comment]
> You want to do the change to move the Start/Submit message evaluation to hooks?

I was planning to do it today - although also would like to start #379, so if you are keen to do it it's fine by me 👍

**Pull request merged**
